mg/target.go

```diff
@@ -0,0 +1,92 @@
+package mg
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+)
+
+// Target checks whether the target files/directories have been more recently
+// modified than src.
+//
+// When recursive is set to true, if src is a directory it will be walked
+// recursively to deermine modified time.
+func Target(src string, recursive bool, targets ...string) (bool, error) {
```

Comment:
The name could be more self helpful, here `Target` could mean the target of an execution or something else. maybe something like `HasBeenModify` or `IsUpToDate` could be more meaningful

Reply:
I think this could do well with being moved to a separate package, to help with namespacing.  Maybe call the package target, and then you could have `target.Dir(path string, prerequisites ....string)` which would be recursive and `target.Path(path string, prerequisites ...string)` which would not be recursive.   This is nicer than a boolean, which I dislike using to change logic of a function because it is obscure when you're reading it. `Target("foo", true, "bar", "baz")` <- when reading this, I don't know what "true" means, unless I got look up the definition of the function.